posthog/api/test/test_query_ws.py

```diff
@@ -0,0 +1,98 @@
+from unittest.mock import patch
+from freezegun import freeze_time
+from channels.testing import WebsocketCommunicator
+from django.test import TransactionTestCase
+from posthog.rate_limit import ClickHouseSustainedRateThrottle
+from asgiref.sync import sync_to_async
+from posthog.models.organization import Organization
+from posthog.models.team import Team
+from posthog.models.user import User
+from posthog.models.project import Project
+import random
+
+
+from posthog.asgi import application
+
+from typing import Optional
+
+import functools
+
+
+class AuthWebsocketCommunicator(WebsocketCommunicator):
+    def __init__(self, application, path, user, *args, **kwargs):
+        super().__init__(self._asgi_with_user(application, user), path, *args, **kwargs)
+
+    @classmethod
+    def _asgi_with_user(cls, asgi_app, user):
+        """
+        Update the scope of an ASGI app such that a particular user
+        is already assumed to have been authenticated.
+        """
+
+        async def app(scope, receive, send):
+            scope["user"] = user
+            return await asgi_app(scope, receive, send)
+
+        functools.update_wrapper(app, asgi_app)
+        return app
+
+
+class TestQueryConsumer(TransactionTestCase):
+    user: Optional[User] = None
+
+    async def _create_user(self):
+        # self.team = self.create_team_with_organization(self.organization)
+        # self.user = self.create_user_with_organization(self.organization)
+
+        if self.user:
+            return self.user
+        org = await Organization.objects.acreate(slug=f"org-{random.randint(1, 1000000)}")
+        project_id = await sync_to_async(Team.objects.increment_id_sequence)()
+        project = await Project.objects.acreate(id=project_id, organization=org)
+        await Team.objects.acreate(organization=org, project=project)
```

Comment:
offtopic: I am slightly worried about this mutual dependency between Team and Project.